### PR TITLE
Revert "Resolve task properties when checking for mutations"

### PR DESCRIPTION
**Pull request by wolfs**
This allows the cache fix plugin to continue working with Gradle 6.8.

This reverts commit c9543ead83fc916d3fbf484d5ef9869d05914202.

### CODE REVIEW

**Comment by wolfs** [comment]
@bot-gradle test this

**Comment by eskatos** [comment]
Fix confirmed at https://github.com/gradle/gradle/issues/15397#issuecomment-743782725

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskExecutionModeExecuter.java [hunk 1]
```diff
@@ -33,24 +37,32 @@
...
+        context.setTaskProperties(properties);
+        TaskExecutionMode taskExecutionMode = executionModeResolver.getExecutionMode(task, properties);
+
         context.setTaskExecutionMode(taskExecutionMode);
         LOGGER.debug("Putting task artifact state for {} into context took {}.", task, clock.getElapsed());
         try {
             return executer.execute(task, state, context);
         } finally {
             context.setTaskExecutionMode(null);
+            context.setTaskProperties(null);
```
Comment (lptr) [inline subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskExecutionModeExecuter.java:65]: I think this is not necessary.

**Review by lptr — COMMENTED** [review]
Looks good in general, I think we should salvage the changes we can instead of completely reverting.

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskStateInternal.java [hunk 2]
```diff
@@ -73,11 +73,11 @@ public void addFailure(TaskExecutionException failure) {
             this.failure = failure;
         } else if (this.failure instanceof TaskExecutionException) {
             TaskExecutionException taskExecutionException = (TaskExecutionException) this.failure;
-            List<Throwable> causes = new ArrayList<>(taskExecutionException.getCauses());
+            List<Throwable> causes = new ArrayList<Throwable>(taskExecutionException.getCauses());
             CollectionUtils.addAll(causes, failure.getCauses());
             taskExecutionException.initCauses(causes);
         } else {
-            List<Throwable> causes = new ArrayList<>();
+            List<Throwable> causes = new ArrayList<Throwable>();
```
[the remark this answers is not in the document]
Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/api/internal/tasks/TaskStateInternal.java:80]: I reverted the commit. I don't plan on polishing the revert. I'll revert the revert on `master`.

File: subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskExecutionModeExecuter.java (hunk 1 above)
Comment (wolfs) [reply subprojects/core/src/main/java/org/gradle/api/internal/tasks/execution/ResolveTaskExecutionModeExecuter.java:65]: I reverted the commit. I don't plan on polishing the revert. I'll revert the revert on `master`.

**Review by lptr — APPROVED** [review]
LGTM.

**Pull request merged**